src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -667,6 +668,62 @@ def create_application(client, display_name, homepage=None, identifier_uris=None
     return result
 
 
+def grant_application(cmd, identifier, appid, expires='1'):
+    if not appid:
+        raise CLIError("'--appid' is required for grant OAuth2 permissions")
+
+    graph_client = _graph_client_factory(cmd.cli_ctx)
+
+    # Get the Service principal ObjectId for the client app
+
+    sps = list(graph_client.service_principals.list(
+        filter="appId eq '{}'".format(identifier)))
+
+    if not sps:
+        raise CLIError("'--id'" + identifier + " does not exist on this directory")
+    client_spn = sps[0].object_id
```

Comment:
Suggest name it `sp_object_id`, as object id is not a service principal name. SPN is either related application's id, or app id uri.

Reply:
Implemented as ```client_sp_object_id```